R/write.R

```diff
@@ -72,18 +72,29 @@ write_csv <- function(x, path, na = "NA", append = FALSE, col_names = !append) {
 
 #' @rdname write_delim
 #' @export
-write_excel_csv <- function(x, path, na = "NA", append = FALSE, col_names = !append) {
+write_excel_csv <- function(x, path, delim = ",", na = "NA", append = FALSE, col_names = !append) {
   stopifnot(is.data.frame(x))
 
-  x_out <- lapply(x, output_column)
-  stream_delim(x_out, path, ",", col_names = col_names, append = append,
+  x_formated <- if(delim == ";") {
+    dplyr::mutate_if(x, is.numeric, dplyr::funs(format(., decimal.mark = ",")))
```

Comment:
hmmmm
so, this function in a hidden way changes the decimal mark based on delim? this may be confusing to people -- what if they only want to change the delim? they might accidentally change the decimal mark, which will be very confusing. I have to alternative ideas:
1. move this conversion to `write_excel_csv2` (or why not?)
2. add a separate argument `decimal.mark` to this function

Reply:
I was thinking about that, but I decided not to ado this because my goal was to create function that would save csv in a strictly defined format -> write_excel_csv -> `.` and `,` and write_excel_csv2 "," and ";". 
I agree that adding this dplyr manipulation the the body of write_excel_csv isn't the best solution, but I didn't have a better idea. I wanted to make minimal changes to  `write_excel_csv` and at the same time didn't want to duplicate the logic in `write_excel_csv2`

Reply:
@marek im open to suggestions how to do it better and move conversion to write_excel_csv2

Reply:
hmm but what if we do the following:
1. keep the delim argument in `write_excel_csv`, but make it only actually change the delimiter and nothing else
2. move this line `dplyr::mutate_if(x, is.numeric, dplyr::funs(format(., decimal.mark = ",")))` to `write_excel_csv2` and pass the result when calling `write_excel_csv`?

this should work unless i'm missing something

Reply:
done PTAL, my only conncern is that `mutate` is done on `x` before checking if its a dataframe. This check is inside `read_excel_csv`. I'm not sure if I should have an additional check before mutate as well?

Reply:
good point. I think we should have this check here as well. I think we can live with duplicated check.